pallets/order-book/src/lib.rs

```diff
@@ -241,28 +247,54 @@ pub mod pallet {
 
     #[pallet::error]
     pub enum Error<T> {
-        /// Trading pair currently reached its capacity
-        OrderLimitReached,
-        /// Price in given order exceeds allowed limits for the trading pair
-        PriceExceedsLimits,
-        /// Lifespan exceeds defined limits
-        InvalidLifespan,
         /// Order book does not exist for this trading pair
         UnknownOrderBook,
         /// Order book already exists for this trading pair
         OrderBookAlreadyExists,
-        /// Cannot insert the limit order because bounds are reached
+        /// Limit order does not exist for this trading pair and order id
+        UnknownLimitOrder,
+        /// Limit order already exists for this trading pair and order id
+        LimitOrderAlreadyExists,
+        /// It is impossible to insert the limit order because the bounds have been reached
         LimitOrderStorageOverflow,
-        /// Cannot delete the limit order
+        /// It is impossible to update the limit order
+        UpdateLimitOrderError,
+        /// It is impossible to delete the limit order
         DeleteLimitOrderError,
+        /// There are no bids/asks for the price
+        NoDataForPrice,
+        /// There are no aggregated bids/asks for the order book
+        NoAggregatedData,
         /// There is not enough liquidity in the order book to cover the deal
         NotEnoughLiquidity,
         /// Cannot create order book with equal base and target assets
         ForbiddenToCreateOrderBookWithSameAssets,
         /// The asset is not allowed to be base. Only dex base asset can be a base asset for order book
         NotAllowedBaseAsset,
         /// User cannot create an order book with NFT if they don't have NFT
-        UserDoesntHaveNft,
+        UserHasNoNft,
+        /// Lifespan exceeds defined limits
+        InvalidLifespan,
+        /// The order amount (limit or market) does not meet the requirements
+        InvalidOrderAmount,
+        /// The limit order price does not meet the requirements
```

Comment:
Maybe also mention that this error is returned when trying to place cross-spreading order and trading is forbidden?
[this case](https://github.com/sora-xor/sora2-network/blob/6eba7aac9790326d5dd2ae881c8387a30e50b173/pallets/order-book/src/order_book.rs#L160)

Reply:
No need. Price rules should be public, if we are talking about users. If you mean developers, I believe they don't need this info here as well

Reply:
So I thought it makes sense to make them clear and exhaustive

Reply:
It is still suitable for blockchain. Invalid price is invalid price